src/tools/w3dviewer/mainfrm.cpp

```diff
@@ -0,0 +1,1607 @@
+{
+    // do later
+}
+
+void CMainFrame::OnRenameAggregate()
+{
+    // do later
+}
+
+void CMainFrame::OnRecordSceneCamera()
+{
+    // do later
+}
+
+void CMainFrame::OnIncludeNull()
+{
+    // do later
+}
+
+void CMainFrame::OnPrevLod()
+{
+    // do later
+}
+
+void CMainFrame::OnNextLod()
+{
+    // do later
+}
+
+void CMainFrame::OnAutoSwitching()
+{
+    // do later
+}
+
+void CMainFrame::OnMovie()
+{
+    // do later
+}
+
+void CMainFrame::OnScreenshot()
+{
+    // do later
+}
+
+void CMainFrame::OnNext()
+{
+    CDataTreeView *view = reinterpret_cast<CDataTreeView *>(m_splitter.GetPane(0, 0));
+
+    if (view != nullptr) {
+        view->SelectNext();
+    }
+}
+
+void CMainFrame::OnPrev()
+{
+    CDataTreeView *view = reinterpret_cast<CDataTreeView *>(m_splitter.GetPane(0, 0));
+
+    if (view != nullptr) {
+        view->SelectPrev();
+    }
+}
+
+void CMainFrame::OnAnimAdvanced()
+{
+    // do later
+}
+
+void CMainFrame::OnCameraSettings()
+{
+    // do later
+}
+
+void CMainFrame::OnCopyScreenSize()
+{
+    // do later
+}
+
+void CMainFrame::OnMissingTextures()
+{
+    // do later
+}
+
+void CMainFrame::OnCopyDeps()
+{
+    // do later
+}
+
+void CMainFrame::OnExposePrecalc()
+{
+    // do later
+}
+
+void CMainFrame::OnTexturePath()
+{
+    // do later
+}
+
+void CMainFrame::OnCreateSphere()
+{
+    // do later
+}
+
+void CMainFrame::OnCreateRing()
+{
+    // do later
+}
+
+void CMainFrame::OnEditPrimitive()
+{
+    // do later
+}
+
+void CMainFrame::OnExportPrimitive()
+{
+    // do later
+}
+
+void CMainFrame::OnKillSceneLight()
+{
+    LightClass *light = GetCurrentDocument()->m_light;
+
+    if (light != nullptr) {
+        light->Set_Diffuse(Vector3(0.0f, 0.0f, 0.0f));
+        light->Set_Specular(Vector3(0.0f, 0.0f, 0.0f));
+    }
+}
+
+void CMainFrame::OnMultiPassLighting()
+{
+    // do later
+}
+
+void CMainFrame::OnMultiTextureLighting()
+{
+    // do later
+}
+
+void CMainFrame::OnVertexLighting()
+{
+    // do later
+}
+
+void CMainFrame::OnAddObject()
+{
+    // do later
+}
+
+void CMainFrame::OnImportFacial()
+{
+    // do later
+}
+
+void CMainFrame::OnRestrictAnims()
+{
+    CDataTreeView *view = reinterpret_cast<CDataTreeView *>(m_splitter.GetPane(0, 0));
+
+    if (view != nullptr) {
+        view->RestrictAnims(!view->m_restrictAnims);
+    }
+}
+
+void CMainFrame::OnBindSubobject()
+{
+    // do later
+}
+
+void CMainFrame::OnSetDistance()
+{
+    // do later
+}
+
+void CMainFrame::OnAlternateMaterials()
+{
+    CW3DViewDoc *document = GetCurrentDocument();
+    document->ToggleAlternateMaterials(nullptr);
+}
+
+void CMainFrame::OnCreateSoundObject()
+{
+    // do later
+}
+
+void CMainFrame::OnEditSoundObject()
+{
+    // do later
+}
+
+void CMainFrame::OnExportSoundObject()
+{
+    // do later
+}
+
+void CMainFrame::OnWireframeMode()
+{
+    CW3DViewDoc *document = GetCurrentDocument();
+    document->m_scene->Set_Polygon_Mode((SceneClass::PolyRenderType)((document->m_scene->Get_Polygon_Mode() == 1) + 1));
```

Comment:
```c++
(document->m_scene->Get_Polygon_Mode() == 1) + 1)
```
reads like something a compiler would write and only works given the specific enum values of 
```c++
    enum PolyRenderType
    {
        POINT = 0x0,
        LINE = 0x1,
        FILL = 0x2,
    };
```
Rewrite as:
```c++
SceneClass::PolyRenderType curType = document->m_scene->Get_Polygon_Mode();
SceneClass::PolyRenderType newType = curType == SceneClass::LINE ? SceneClass::FILL : SceneClass::LINE;
document->m_scene->Set_Polygon_Mode(newType);
```

Reply:
fixed.